pkg/controller/common/reconciler/reconciler.go

```diff
@@ -111,9 +111,38 @@ func ReconcileResource(params Params) error {
 			return err
 		}
 		return nil
+	}
+
+	// Check if already exists
+	err = params.Client.Get(types.NamespacedName{Name: name, Namespace: namespace}, params.Reconciled)
+	if err != nil && apierrors.IsNotFound(err) {
+		return create()
 	} else if err != nil {
 		log.Error(err, fmt.Sprintf("Generic GET for %s %s/%s failed with error", kind, namespace, name))
-		return errors.Wrapf(err, "failed to get %s %s/%s", kind, namespace, name)
+		return fmt.Errorf("failed to get %s %s/%s: %w", kind, namespace, name, err)
+	}
+
+	if params.NeedsRecreate != nil && params.NeedsRecreate() {
+		log.Info("Resource cannot be updated, hence will be deleted and then recreated", "kind", kind, "namespace", namespace, "name", name)
```

Comment:
Nit: it seems like we can consolidate these two info messages

Reply:
I think it's fine to keep it separate, so we can search for `"Deleting resource"` or `"Creating resource"` in logs to see all created resources, this one not being an exception.